Sources/Brave/Frontend/Player/Player.html

```diff
@@ -0,0 +1,26 @@
+<!DOCTYPE html>
+<!-- Copyright 2023 The Brave Authors. All rights reserved.
+  -- This Source Code Form is subject to the terms of the Mozilla Public
+  -- License, v. 2.0. If a copy of the MPL was not distributed with this
+  -- file, You can obtain one at https://mozilla.org/MPL/2.0/.
+  -->
+<html lang="en">
+
+<html>
+  <head>
+    <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1, user-scalable=no, viewport-fit=cover">
+    <link rel="stylesheet" type="text/css" href="/styles/Player.css">
+    <script src="/scripts/Player.js"></script>
+    <title>Brave Player</title>
+  </head>
+  <body>
+    <div id="player-container">
+      <iframe id="player" width="100%" src="https://www.youtube-nocookie.com/embed/%YOUTUBE_VIDEO_ID%" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share" allowfullscreen></iframe>
```

Comment:
per https://bravesoftware.slack.com/archives/C0YL5KMA8/p1700000789276839, please add an automated or QA test to ensure that Brave Shields content scripts are running in this iframe.

Reply:
This is going to be very tricky to do, but I don't think it's impossible. But it will require some investigation and trickery on my part to make this work

Reply:
@diracdeltas I thought about this. Is this a blocking request? I'm not sure how to approach this yet and I don't think I can accomplish anything in a reasonable amount of time. I'd like to push this off as a subtask.

Reply:
Also a note on this. The brave player acts like any other page. Meaning you can disable shields on it via the shields button. And shields essentially work as any other page here. I don't know if this is expected but the figma file has the shields icon on the navigation bar so one can only assume that shields can be disabled on this page. Plus I would really like to avoid any specific shield handling for this page if possible.

Since this brave player loads like any other page, adding tests here shouldn't be any different than any other shields tests we may already have or need to build. So I don't think such tests should be part of this task. 

Unfortunately we don't have any navigation delegate tests. Navigation delegate is responsible for managing (adding removing, etc) scriplets and content blockers so it seems fit to write out tests for the navigation delegate which could include any specific url tests like the brave player but is not really a task for the brave player. This probably would require some code reshuffling to make this work and will surely bloat any PR with many code changes as functions and classes are reshuffled for some greater modularity.

Reply:
Just to confirm, Shields (the button) works on brave://player and all other internal URLs in iOS. For things like reader mode, if you click the Shields button, it will show the domain from which reader mode extracted the text, but in fact it runs in internal://local context, we just show the original domain:
<img src="https://github.com/brave/brave-ios/assets/10810135/a2f9484a-3f89-4d8b-9bb3-ec4104cea3de" width=300>
In case of player, I see it blocks some XHR to `https://www.youtube-nocookie.com/youtubei/v1/log_event?alt=json&key=AIzaSyAO_FJ2SlqU8Q4STEHLGCilw_Y9_11qcW8` and `https://play.google.com/log?format=json&hasfast=true&authuser=0` but the counter at brave://player stays at 0 while if you go directly to youtube-nocookie.com it goes up to 1:
<img src='https://github.com/brave/brave-ios/assets/10810135/0908ddc3-00cf-40ed-ad8d-0750b6cfafb9' width=300>
<img src='https://github.com/brave/brave-ios/assets/10810135/74420c27-a75b-4cf6-9465-ee1077826f01' width=300>

Not sure if the counter is off or something else

Reply:
@cuba ok to just add a QA test case for it for now instead of automated

@stoletheminerals is it a security issue that shields can run on internal:// origins? my main concern is that if these origins have special privileges (similar to brave:// on desktop), a malicious uBlock scriptlet or something that is injected as part of shields could do more damage than usual.

Reply:
@stoletheminerals I can take a look at why the counter is not increasing. I don't remember if we intentionally only log main frame blocks in iOS. Perhaps we do this accidentally. But I would say this is a separate issue.

Reply:
>@stoletheminerals is it a security issue that shields can run on internal:// origins? my main concern is that if these origins have special privileges (similar to brave:// on desktop), a malicious uBlock scriptlet or something that is injected as part of shields could do more damage than usual.

@diracdeltas it's different than brave:// or chrome:// on Desktop, internal:// URLs on iOS don't have any specific privileges compared to any other web page. But let me think, maybe I can find some edge cases.

Reply:
@diracdeltas internal pages we have are:

- Interstitials for SSL errors and other error pages (404, etc)
- Web3 interstitials for ENS, SNS, IPFS resolution (consent interstitials basically)
- Session restore page for history restoring
- Reader mode
- Player

I don't see any immediate issues with Shields running on these pages (maybe clicking `Proceed` on a spoofed SSL certificate interstitial, but I guess a malicious script can do something worse in the context of the domain). These pages don't have any specific direct communication mechanisms to Native side, only webkit message handlers which are available on normal pages as well and we secured those as much as possible.
We have Wallet WebUI on iOS coming at some point, there is no final implementation yet, but Shields definitely should not be running there. I filled [this issue ](https://github.com/brave/brave-ios/issues/8441)to disable Shields in non-http schemes, I think it's fine to work on it separately from this PR, Player's internal page is the least concerning anyway.

Reply:
sgtm, thanks!